system/lib/wasmfs/memory_backend.h

```diff
@@ -25,6 +25,9 @@ class MemoryDataFile : public DataFile {
   int flush() override { return 0; }
   off_t getSize() override { return buffer.size(); }
   int setSize(off_t size) override {
+    if (size > buffer.max_size()) {
+      return -EOVERFLOW;
+    }
```

Comment:
@tlively how could we have detected this bug? (i.e. the silent truncation of `off_t` (uint64_t) to `size_t` (uint32_t)) ?

Would building libwasmfs with UBSAN have done it?

Reply:
Opened https://github.com/emscripten-core/emscripten/issues/24261